.github/workflows/tests_entry.yml

```diff
@@ -0,0 +1,95 @@
+# tests_entry.yml
+# Entry point to the test suites
+#
+# - If neither ci:run or ci:ready_to_merge labels are on PR, fail and exit.
+# - If ci:run label is on PR, remove ci:run, run the test scripts.
+# - If ci:ready_to_merge is on the PR and the action is synchronize not triggered
+#   by the mergify[bot] user, remove label and fail job.
+# - If ci:ready_to_merge label is on PR and action is labeled, run the test scripts.
+# 
+# The end result is labeling ci:run or ci:ready_to_merge will run the test scripts,
+# If Mergify merges to the PR, the test scripts will run. If anyone else tries to add 
+# a commit to the PR or merge , the script will fail and ci:ready_to_merge will be
+# removed.
+#
+# This script runs the test scripts directly. Scheduled or manual runs use 
+# run_<scriptname>.yml as the entry point. 
+
+name: Tests Entry Point
+on:
+  pull_request_target:
+    types:
+      - synchronize
+      - labeled
+
+jobs:
+  no-labels:
+    runs-on: ubuntu-latest
+    if: ${{ !(contains(github.event.pull_request.labels.*.name, 'ci:run') ||
+        contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge')) }}
+    steps:
+      - run: exit 1
+  some-labels:
+    runs-on: ubuntu-latest
+    if: ${{ (contains(github.event.pull_request.labels.*.name, 'ci:run') ||
+        contains(github.event.pull_request.labels.*.name, 'ci:ready_to_merge')) }}
```

Comment:
Eh, turns out this isn't as clean as we would like. Putting the `if` condition at the job level means the job is skipped entirely if not met. Any jobs after that will skip if they have a `needs` on that job. So while we can separate each label conceptually this way, we still need a little bit of ugly if logic in the steps. Still doing it, but it isn't as pretty as your code above would have been if the skip behavior would have been different.
And using the `if: always()` dodge to make sure the following job runs even if the previous job skips won't work because that will trap our exits and make sure our tests will always run no matter what.

Reply:
understood. perhaps we should make some environment variables for the different expressions?

https://docs.github.com/en/actions/learn-github-actions/expressions#example-setting-an-environment-variable

```yaml
env:
  SYNC_BY_MERGIFY: ${{ (github.event.action == 'synchronize') && !(github.event.sender.login == 'mergify[bot]') }}
```